packages/devtools_app/lib/src/flutter/extent_delegate_list.dart

```diff
@@ -0,0 +1,464 @@
+        ++index) {
+      RenderBox child = childAfter(trailingChildWithLayout);
+      if (child == null || indexOf(child) != index) {
+        child = insertAndLayoutChild(buildChildConstraints(index),
+            after: trailingChildWithLayout);
+        if (child == null) {
+          // We have run out of children.
+          estimatedMaxScrollOffset = _extentDelegate.layoutOffset(index + 1);
+          break;
+        }
+      } else {
+        child.layout(buildChildConstraints(index));
+      }
+      trailingChildWithLayout = child;
+      assert(child != null);
+      final SliverMultiBoxAdaptorParentData childParentData =
+          child.parentData as SliverMultiBoxAdaptorParentData;
+      assert(childParentData.index == index);
+      childParentData.layoutOffset =
+          _extentDelegate.layoutOffset(childParentData.index);
+    }
+
+    final int lastIndex = indexOf(lastChild);
+    final double leadingScrollOffset = _extentDelegate.layoutOffset(firstIndex);
+    final double trailingScrollOffset =
+        _extentDelegate.layoutOffset(lastIndex + 1);
+
+    assert(firstIndex == 0 ||
+        childScrollOffset(firstChild) - scrollOffset <=
+            precisionErrorTolerance);
+    assert(debugAssertChildListIsNonEmptyAndContiguous());
+    assert(indexOf(firstChild) == firstIndex);
+    assert(targetLastIndex == null || lastIndex <= targetLastIndex);
+
+    estimatedMaxScrollOffset = math.min(
+      estimatedMaxScrollOffset,
+      estimateMaxScrollOffset(
+        constraints,
+        firstIndex: firstIndex,
+        lastIndex: lastIndex,
+        leadingScrollOffset: leadingScrollOffset,
+        trailingScrollOffset: trailingScrollOffset,
+      ),
+    );
+
+    final double paintExtent = calculatePaintOffset(
+      constraints,
+      from: leadingScrollOffset,
+      to: trailingScrollOffset,
+    );
+
+    final double cacheExtent = calculateCacheOffset(
+      constraints,
+      from: leadingScrollOffset,
+      to: trailingScrollOffset,
+    );
+
+    final double targetEndScrollOffsetForPaint =
+        constraints.scrollOffset + constraints.remainingPaintExtent;
+    final int targetLastIndexForPaint = targetEndScrollOffsetForPaint.isFinite
+        ? _extentDelegate
+            .maxChildIndexForScrollOffset(targetEndScrollOffsetForPaint)
+        : null;
+    geometry = SliverGeometry(
+      scrollExtent: estimatedMaxScrollOffset,
+      paintExtent: paintExtent,
+      cacheExtent: cacheExtent,
+      maxPaintExtent: estimatedMaxScrollOffset,
+      // Conservative to avoid flickering away the clip during scroll.
+      hasVisualOverflow: (targetLastIndexForPaint != null &&
+              lastIndex >= targetLastIndexForPaint) ||
+          constraints.scrollOffset > 0.0,
+    );
+
+    // We may have started the layout while scrolled to the end, which would not
+    // expose a new child.
+    if (estimatedMaxScrollOffset == trailingScrollOffset)
```

Comment:
nit: add brackets to make this more readable

Reply:
done